# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.h
# Commit: d9efa80409ee
# Full Hash: d9efa80409eed34d06fb989e6b5ebea0f58bd60f
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-11-10 09:24:53
# Regressor Bug: 1657375
# File Overlap Count: 1
# Description:
#   Bug 1657375 - Cache the principal hash value for OffscreenCanvas on worker threads. r=dom-worker-reviewers,asuth,smaug
#   
#   We cannot access ClientWebGLContext::mCanvasElement or its associated
#   nsIPrincipal off the main thread. We use the hash value of the principal
#   to limit how many WebGL contexts a single domain can create. We can
# ==============================================================================

diff -r 2f9159719ec5 -r d9efa80409ee dom/workers/WorkerPrivate.h
--- a/dom/workers/WorkerPrivate.h	Tue Nov 09 20:16:04 2021 +0000
+++ b/dom/workers/WorkerPrivate.h	Tue Nov 09 20:16:05 2021 +0000
@@ -739,6 +739,10 @@
     return *mLoadInfo.mPrincipalInfo;
   }
 
+  uint32_t GetPrincipalHashValue() const {
+    return mLoadInfo.mPrincipalHashValue;
+  }
+
   const mozilla::ipc::PrincipalInfo& GetEffectiveStoragePrincipalInfo() const;
 
   already_AddRefed<nsIChannel> ForgetWorkerChannel() {